napari/plugins/_npe2.py

```diff
@@ -201,7 +201,7 @@ def get_readers(path: str) -> Dict[str, str]:
     pm = npe2.PluginManager.instance()
     return {
         pm.get_manifest(reader.command).display_name: reader.plugin_name
-        for reader in pm.iter_compatible_readers(path)
+        for reader in pm.iter_compatible_readers([path])
```

Comment:
just a note for myself here... the current implementation of `pm.iter_compatible_readers` is:

```python
    def iter_compatible_readers(
        self, path: Union[PathLike, List[PathLike]]
    ) -> Iterator[ReaderContribution]:
        if not path:
            return

        if isinstance(path, list):
            if len({Path(i).suffix for i in path}) > 1:
                raise ValueError(
                    "All paths in the stack list must have the same extension."
                )
            path = path[0]
```

So this will indeed lead to the exact same result for now (but we shouldn't forget that it will only take the first item in the list until that is changed upstream)